src/Modules/Settings/Dnn.PersonaBar.Prompt/Components/ModulesController.cs

```diff
@@ -113,7 +113,13 @@ public ModuleInfo CopyModule(PortalSettings portalSettings, int moduleId, int so
                 return null;
             }
 
-            if (targetPage.PortalID == portalSettings.PortalId  || PortalHelper.IsRequestForSiteGroup(targetPage.PortalID,portalSettings.PortalId))
+            var currentPortalSetting = PortalController.Instance.GetCurrentPortalSettings();
+
+            if (
+                currentPortalSetting == portalSettings ||
```

Comment:
can we compare portal id instead of settings here?

Reply:
No way, as I commented above that was much needed did enable us to differentiate a request for portalId either passed for cross-scripting or not. And it really helped to understand if for example, host user has token for site3 but not passing the crossportalid, and trying to access module for site1, which he should get and this check enabling him and if same time he passed the portal id 3 then he should not get.